aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -2165,5 +2166,21 @@ Tensor sum_sparse_coo(const Tensor& self, at::OptionalIntArrayRef dim, bool keep
   return result;
 }
 
+Tensor sum_sparse_csr(const Tensor& self, at::OptionalIntArrayRef dim, bool keepdim, c10::optional<ScalarType> dtype) {
+  // TODO: The signature of sum.dim_IntList and _sparse_csr_sum.dim_dtype is a little
+  // bit different in the second parameters `dim`, which causes the conversion of `dim`
+  // to call into `_sparse_csr_sum`. Align the signatures would be a better choice.
+  TORCH_CHECK(dim.has_value(),"dim has no value, cannot be used in sum.dim_IntList");
+  if (self.is_sparse_csr()) {
+    return at::_sparse_csr_sum(self, *dim, true, dtype);
+  } else if (self.layout() == kSparseCsc) {
+    Tensor new_self = self.to_dense().to_sparse_csr();
+    return at::_sparse_csr_sum(new_self, *dim, true, dtype);
```

Comment:
Conversion to a strided tensor is unnecessary, using `self.to_sparse_csr()` should be sufficient and is more efficient. That said, I think no conversions are necessary because we have the following invariants:
```
batch_dim = csc.dim() - csc.dense_dim() - csc.sparse_dim()
csc.layout == torch.sparse_csc
csc.transpose(batch_dim, batch_dim + 1).layout == torch.sparse_csr
torch.sum(csc, dim=dim, keepdim=True) == torch.sum(csc.transpose(batch_dim, batch_dim+1),
      dim=(*dim[:batch_dim], dim[batch_dim+1], dim[batch_dim], *dim[batch_dim+2:]),
      keepdim=True).transpose(batch_dim, batch_dim+1)
```
So, I suggest using the following method for computing reductions on a CSC tensor:
```suggestion
    auto batch_dim = csc.dim() - csc.dense_dim() - csc.sparse_dim();
    auto swapped_dim = ...;  // a copy of `*dim` where `batch_dim` and `batch_dim+1`- th elements are swapped
    return at::_sparse_csr_sum(self.transpose(batch_dim, batch_dim+1), swapped_dim, true, dtype).transpose(batch_dim, batch_dim+1);
```